python/idsse_common/test/test_validate_schema.py

```diff
@@ -621,16 +666,93 @@ def test_validate_event_port_message_with_missing_metadata(event_port_validator:
+    try:
+        new_data_validator.validate(new_field_message)
+    except ValidationError as exc:
+        assert False, f'Validate message raised an exception {exc}'
 
-    schema_name = 'new_data_schema'
-    validator = get_validator(schema_name)
+
+def test_validate_new_missing_field_data_message(new_data_validator: Validator,
+                                                 new_field_message: dict):
+    # convert a new field message to a missing field message by changing 'field' key to 'missing'
+    new_field_message['missing'] = new_field_message.pop('field')
     try:
-        validator.validate(message)
+        new_data_validator.validate(new_field_message)
     except ValidationError as exc:
         assert False, f'Validate message raised an exception {exc}'
+
+
+def test_validate_new_field_data_message_missing_region(new_data_validator: Validator,
+                                                        new_field_message: dict):
+    new_field_message.pop('region')
+    with raises(ValidationError):
+        new_data_validator.validate(new_field_message)
+
+
+def test_validate_new_valid_data_message(new_data_validator: Validator,
+                                         new_valid_message: dict):
+    try:
+        new_data_validator.validate(new_valid_message)
+    except ValidationError as exc:
+        assert False, f'Validate message raised an exception {exc}'
+
+
+def test_validate_new_missing_valid_data_message(new_data_validator: Validator,
+                                                 new_valid_message: dict):
+    # convert new valid to missing valid by putting a field in missing list
+    fields = new_valid_message['field']
+    new_valid_message['field'] = fields[:-1]
+    new_valid_message['missing'] = [fields[-1]]
+    try:
+        new_data_validator.validate(new_valid_message)
+    except ValidationError as exc:
+        assert False, f'Validate message raised an exception {exc}'
+
+
+def test_validate_new_valid_data_message_bad_field(new_data_validator: Validator,
+                                                   new_valid_message: dict):
+    new_valid_message['field'].append('BAD_FIELD_NAME')
+    with raises(ValidationError):
+        new_data_validator.validate(new_valid_message)
+
+
+def test_validate_new_issue_data_message(new_data_validator: Validator,
+                                         new_issue_message: dict):
+    try:
+        new_data_validator.validate(new_issue_message)
+    except ValidationError as exc:
+        assert False, f'Validate message raised an exception {exc}'
+
+
+def test_validate_new_missing_issue_data_message(new_data_validator: Validator,
+                                                 new_issue_message: dict):
+    # convert new issue to missing issue by putting some fields in missing object
+    fields = new_issue_message['field']
+    found = {}
+    missing = {}
+    for valid_key in fields:
+        if random.randint(0, 1):
```

Comment:
This is a simple, creative solution to add some unpredictability to the test (the good type of unpredictability).